python/mxnet/ndarray/ndarray.py

```diff
@@ -155,6 +155,11 @@ def _new_alloc_handle(shape, ctx, delay_alloc, dtype=mx_real_t):
             ctypes.c_int(int(_DTYPE_NP_TO_MX[np.dtype(dtype).type])),
             ctypes.byref(hdl)))
     else:
+        size = 1
+        for idx in shape:
+            size = size * idx
+        if size > 2**32:
```

Comment:
This also happens for `size < 2**32`. For example with the matrix of shape `(18545452, 200)` in https://github.com/dmlc/gluon-nlp/issues/981.

Please make the check more accurate

Reply:
@leezu I have fixed that. Please see the output below and verify.
`
ubuntu@ip-172-31-82-110 ~/fast_text $ python code.py
/home/ubuntu/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Traceback (most recent call last):
  File "code.py", line 4, in <module>
    model = nlp.model.train.FasttextEmbeddingModel.load_fasttext_format(wordvectors)
  File "/home/ubuntu/anaconda3/lib/python3.6/site-packages/gluonnlp/model/train/embedding.py", line 277, in load_fasttext_format
    self.initialize(ctx=ctx)
  File "/home/ubuntu/incubator-mxnet/python/mxnet/gluon/block.py", line 632, in initialize
    self.collect_params().initialize(init, ctx, verbose, force_reinit)
  File "/home/ubuntu/incubator-mxnet/python/mxnet/gluon/parameter.py", line 887, in initialize
    v.initialize(None, ctx, init, force_reinit=force_reinit)
  File "/home/ubuntu/incubator-mxnet/python/mxnet/gluon/parameter.py", line 464, in initialize
    self._finish_deferred_init()
  File "/home/ubuntu/incubator-mxnet/python/mxnet/gluon/parameter.py", line 344, in _finish_deferred_init
    data = zeros_fn(**kwargs)
  File "/home/ubuntu/incubator-mxnet/python/mxnet/ndarray/utils.py", line 67, in zeros
    return _zeros_ndarray(shape, ctx, dtype, **kwargs)
  File "/home/ubuntu/incubator-mxnet/python/mxnet/ndarray/ndarray.py", line 4641, in zeros
    return _internal._zeros(shape=shape, ctx=ctx, dtype=dtype, **kwargs)
  File "<string>", line 36, in _zeros
  File "/home/ubuntu/incubator-mxnet/python/mxnet/_ctypes/ndarray.py", line 107, in _imperative_invoke
    ctypes.byref(out_stypes)))
  File "/home/ubuntu/incubator-mxnet/python/mxnet/base.py", line 254, in check_call
    raise MXNetError(py_str(_LIB.MXGetLastError()))
mxnet.base.MXNetError: [00:25:17] src/operator/numpy/../tensor/init_op.h:276: Check failed: param_shape.Size() < (int64_t{1} << 31) - 1 (3709090400 vs. 2147483647) : [InitShape-input] Size of tensor you are trying to allocate is larger than 2^31 elements. Please build with flag USE_INT64_TENSOR_SIZE=1
Stack trace:
  [bt] (0) /home/ubuntu/incubator-mxnet/python/mxnet/../../lib/libmxnet.so(dmlc::LogMessageFatal::~LogMessageFatal()+0x34) [0x7efc0bd7df44]
  [bt] (1) /home/ubuntu/incubator-mxnet/python/mxnet/../../lib/libmxnet.so(bool mxnet::op::InitShape<mxnet::op::InitOpParam>(nnvm::NodeAttrs const&, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*)+0x339) [0x7efc0ca83045]
  [bt] (2) /home/ubuntu/incubator-mxnet/python/mxnet/../../lib/libmxnet.so(std::_Function_handler<bool (nnvm::NodeAttrs const&, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*), bool (*)(nnvm::NodeAttrs const&, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*)>::_M_invoke(std::_Any_data const&, nnvm::NodeAttrs const&, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*&&, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*&&)+0x61) [0x7efc0bd96b9b]
  [bt] (3) /home/ubuntu/incubator-mxnet/python/mxnet/../../lib/libmxnet.so(std::function<bool (nnvm::NodeAttrs const&, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*)>::operator()(nnvm::NodeAttrs const&, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*, std::vector<mxnet::TShape, std::allocator<mxnet::TShape> >*) const+0x76) [0x7efc1053c68e]
  [bt] (4) /home/ubuntu/incubator-mxnet/python/mxnet/../../lib/libmxnet.so(mxnet::imperative::SetShapeType(mxnet::Context const&, nnvm::NodeAttrs const&, std::vector<mxnet::NDArray*, std::allocator<mxnet::NDArray*> > const&, std::vector<mxnet::NDArray*, std::allocator<mxnet::NDArray*> > const&, mxnet::DispatchMode*)+0x4c5) [0x7efc106a42b2]
  [bt] (5) /home/ubuntu/incubator-mxnet/python/mxnet/../../lib/libmxnet.so(mxnet::Imperative::Invoke(mxnet::Context const&, nnvm::NodeAttrs const&, std::vector<mxnet::NDArray*, std::allocator<mxnet::NDArray*> > const&, std::vector<mxnet::NDArray*, std::allocator<mxnet::NDArray*> > const&)+0x285) [0x7efc106aadf9]
  [bt] (6) /home/ubuntu/incubator-mxnet/python/mxnet/../../lib/libmxnet.so(MXImperativeInvokeImpl(void*, int, void**, int*, void***, int, char const**, char const**)+0x195) [0x7efc11007089]
  [bt] (7) /home/ubuntu/incubator-mxnet/python/mxnet/../../lib/libmxnet.so(MXImperativeInvokeEx+0x87) [0x7efc110073fa]
  [bt] (8) /home/ubuntu/anaconda3/lib/python3.6/lib-dynload/../../libffi.so.6(ffi_call_unix64+0x4c) [0x7efc1a86eec0]
`